libtelemetry/src/full/java/com/mapbox/android/telemetry/MapboxTelemetry.java

```diff
@@ -412,13 +413,34 @@ public void run() {
     });
   }
 
-  private static synchronized void setAccessToken(@NonNull Context context, @NonNull String accessToken) {
+  private static synchronized void setAccessToken(@NonNull final Context context,
+                                                  @NonNull String accessToken,
+                                                  @NonNull ExecutorService executorService) {
     if (TelemetryUtils.isEmpty(accessToken)) {
       return;
     }
     if (sAccessToken.getAndSet(accessToken).isEmpty()) {
+      sendErrorReports(context, executorService);
+    }
+  }
+
+  private static void sendErrorReports(@NonNull final Context context,
+                                       @NonNull final ExecutorService executorService) {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
       LocalBroadcastManager.getInstance(context)
         .sendBroadcast(new Intent(MapboxTelemetryConstants.ACTION_TOKEN_CHANGED));
+    } else {
+      try {
+        executorService.execute(new Runnable() {
+          @Override
+          public void run() {
+            CrashReporter crashReporter = new CrashReporter(context);
+            crashReporter.sendErrorReports();
+          }
+        });
+      } catch (Throwable throwable) {
```

Comment:
Might deserve a note that you are catching Throwable because we don't want the crash reporter to report it's own reporting errors:


![](https://i.imgflip.com/3vr448.jpg)